general/auth/2fa.rst

```diff
@@ -0,0 +1,127 @@
+
+Requirements
+============
+
+.. note:: These lists are just examples, they are not endorsements of
+          any specific software.
+
+If you don't already have one, you will need to choose an
+authenticator.
+
+Phone-based authenticators are the easiest and most common so we will
+assume you'll pick and install one on your phone, examples include
+`authy <https://authy.com/>`_, `freeotp
+<https://freeotp.github.io/>`_, `google authenticator
+<https://support.google.com/accounts/answer/1066447?hl=en>`_,
+`lastpass authenticator <https://lastpass.com/auth/>`_, `microsoft
+authenticator
+<https://www.microsoft.com/en-gb/account/authenticator?cmp=h66ftb_42hbak>`_,
+...; password managers also commonly include :abbr:`2FA (two-factor
+authentication)` support e.g. `1password
+<https://support.1password.com/one-time-passwords/>`_, `bitwarden
+<https://bitwarden.com/help/article/authenticator-keys/>`_, ...
+
+For the sake of demonstration we will be using Google Authenticator
+(not because it is any good but because it is quite common).
+
+Setting up two-factor authentication
+====================================
+
+Once you have your authenticator of choice, go to the Odoo instance
+you want to setup :abbr:`2FA (two-factor authentication)`, then open
+:guilabel:`Preferences` (or :guilabel:`My Profile`):
+
+.. figure:: media/totp_setup/preferences.png
+    :align: center
+
+Open the :guilabel:`Account Security` tab, then click the
+:guilabel:`Enable two-factor authentication` button:
+
+.. figure:: media/totp_setup/sec_tab.png
+   :align: center
+
+Because this is a security-sensitive action, you will need to input
+your password:
+
+.. figure:: media/totp_setup/sec_enhanced.png
+   :align: center
+
+After which you will see this screen with a barcode:
+
+.. figure:: media/totp_setup/totp_scan.png
+   :align: center
+
+In most applications, you can simply *scan the barcode* via the
+authenticator of your choice, the authenticator will then take care of
+all the setup:
+
+.. figure:: media/totp_setup/scan_barcode.jpg
+   :align: center
+
+.. note::
+
+   If you can not scan the screen (e.g. because you are doing this
+   set-up on the same phone as the authenticator application), you can
+   click the provided link, or copy the secret to manually set-up your
+   authenticator:
+
+   .. figure:: media/totp_setup/secret_visible.png
+      :align: center
+
+   .. figure:: media/totp_setup/input_secret.png
+      :align: center
+
+Once this is done, the authenticator should display a *verification
+code* with some useful identifying information (e.g. the domain and
+login for which the code is):
+
+.. figure:: media/totp_setup/authenticator.png
+   :align: center
+
+You can now input the code into the :guilabel:`Verification Code`
+field, then click the :guilabel:`Enable two-factor authentication`
+button.
+
+Congratulation, your account is now protected by two-factor
+authentication!
+
+.. figure:: media/totp_setup/totp_enabled.png
+   :align: center
```

Comment:
I would add a link to the user guides of the authentication apps,
in the case our documentation would become outdated, the customer can still fall back on these.

Reply:
And while the lastpass authenticator guide still exists, apparently it moved around and half the links are dead, so that seems... less than reliable.

Authy apparently perfers providing [per-application guides](https://authy.com/guides/).